airbyte-integrations/connectors/source-gcs/source_gcs/stream.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import traceback
+from typing import Any, Iterable
+
+from airbyte_cdk.models import AirbyteLogMessage, AirbyteMessage, Level
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.file_based.exceptions import (
+    FileBasedSourceError,
+    MissingSchemaError,
+    RecordParseError,
+    StopSyncPerValidationPolicy,
+)
+from airbyte_cdk.sources.file_based.stream import DefaultFileBasedStream
+from airbyte_cdk.sources.file_based.types import StreamSlice
+from airbyte_cdk.sources.utils.record_helper import stream_data_to_airbyte_message
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from source_gcs.helpers import GCSRemoteFile
+
+
+class GCSStream(DefaultFileBasedStream):
+
+    """
+    Ads transform_record() to base read_records_from_slice().
+
+    """
+
+    def transform_record(self, record: dict[str, Any], file: GCSRemoteFile) -> dict[str, Any]:
+        file_datetime_string = file.last_modified.strftime(self.DATE_TIME_FORMAT)
+
+        record[self.ab_last_mod_col] = file_datetime_string
+        record[self.ab_file_name_col] = file.displayed_uri if file.displayed_uri else file.uri
+        return record
+
+    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[AirbyteMessage]:
```

Comment:
I'm not into the idea of copypaste the whole method to change 1 line only.
it feels like we need to refactor this method in CDK to pass custom `transform_record`

Reply:
created pr for cdk https://github.com/airbytehq/airbyte/pull/45698